### ISSUE TITLE
EnumField does not implement elementContainer and so its Elements are missed in Walk

### ISSUE THREAD

**Issue by jgiles** [emicklei/proto#140]
# Expected behavior

If there is an option on an enum field, like such:

```proto
import "google/api/visibility.proto";
enum MyEnum {
    UNKNOWN = 0;
    PRIVATE = 1 [(google.api.value_visibility).restriction = "HIDDEN"];
}
```

I should be able to process that option via a call like

```golang
proto.Walk(parsedProto, proto.WithOption(myOptionHandler))
```

# Actual behavior
`myOptionHandler` is never called with the option on the `PRIVATE` enum field.

# Suggested root cause and fix
- `EnumField` should implement the `elementContainer` interface, and commentary in [enum.go](https://github.com/emicklei/proto/blob/4ced960506455acb8c53cbdff01c2e548123771b/enum.go#L203) suggests that is the intent
- However, `EnumField` is missing interface methods `elements` and `takeLastComment`
- Because of this, the `walk` recursive traversal [does not identify EnumField as an elementContainer](https://github.com/emicklei/proto/blob/4ced960506455acb8c53cbdff01c2e548123771b/walk.go#L39) and so does not walk its children (including options)

This issue is impacting usecases like [stackb/rules_proto](https://github.com/stackb/rules_proto/blob/52fcb49627defce77f8a18ea837ed1fb3b165951/pkg/protoc/file.go#L146) - @pcj as an fyi

I'm happy to put up a PR adding the missing methods, but am unsure of the correct implementation for `takeLastComment`

**Comment by emicklei** [comment]
thank you for reporting this issue.
because you have doubts about the the takeLastComment, I will have a go at composing the PR

**Comment by jgiles** [comment]
According to my testing this was fixed in #141
